### ISSUE TITLE
pjsua_aud_channel_update for a second call returns PJ_EINVALIDOP

### ISSUE THREAD

**Issue by razvancrainea** [pjsip/pjproject#3324]
### Describe the bug

I am writing an app using the C++ wrapper that is able to place multiple SIP calls. I am running it in a docker container, on a server that does not have sound capabilities.
If I am running one single call, everything works fine. However, if I start a second call a few seconds after the first one is up and running, the calls starts fine, but when the 200 OK is received, the media for second call fails to create, with the following error:
```
pjsua_media.c  ......pjsua_aud_channel_update() failed for call_id 1 media 0: Invalid operation (PJ_EINVALIDOP)
```
I am not sure whether I am doing something wrong, or whether I need to tune something else to get this working, so any advice is welcome.

### Steps to reproduce

1. compile pjsua
2. write an app that starts two calls, with a bit of a delay between them
3. run the app

A small snippet of the app is:
```
// in main
MyCall *call1 = new MyCall(acc, destination1);
CallOpParam prm(true);
call1->makeCall(call1->destination, prm);
MyCall *call2 = new MyCall(acc, destination2);
ep.utilTimerSchedule(5000, call2);

// and in the onTimer() function
CallOpParam prm(true);
call->makeCall(call->destination, prm);
```

I can provide the entire file if needed.

### PJSIP version

2.13

### Context

- Running on Debian bullseye
- compiled using
```
    ./configure --enable-shared \
                --disable-opencore-amr \
                --disable-resample \
                --disable-sound \
                --disable-video \
                --with-external-gsm \
                --with-external-pa \
                --with-external-speex \
                --prefix=/usr
```

### Log, call stack, etc

```shell
14:08:26.784      inv0x55db551e6e38  ....SDP negotiation done: Success
14:08:26.784          pjsua_media.c  .....Call 0: updating media..
14:08:26.785          pjsua_media.c  .......Media stream call00:0 is destroyed
14:08:26.785      udp0x55db551e8230  ......UDP media transport started
14:08:26.785            pjsua_aud.c  ......Audio channel update..
14:08:26.785     strm0x7f40f400c398  .......VAD temporarily disabled
14:08:26.786      udp0x55db551e8230  .......UDP media transport attached
14:08:26.786     strm0x7f40f400c398  .......Encoder stream started
14:08:26.786     strm0x7f40f400c398  .......Decoder stream started
14:08:26.786          pjsua_media.c  ......Audio updated, stream #0: speex (sendrecv)
...
14:08:27.412     strm0x7f40f400c398  VAD re-enabled
14:08:27.512            pjsua_aud.c !Closing sound device after idle for 1 second(s)
14:08:27.512            pjsua_aud.c  .Closing null sound device..
14:08:29.660           pjsua_call.c  Making call with acc #0 to sip:*552023@testing.siphub.net
14:08:29.661            pjsua_aud.c  .Set sound device: capture=-99, playback=-99, mode=0
14:08:29.661            pjsua_aud.c  ..Null sound device, mode setting is ignored
14:08:29.661            pjsua_aud.c  ..Setting null sound device..
14:08:29.661            pjsua_aud.c  ...Opening null sound device..
14:08:29.661          pjsua_media.c  .Call 1: initializing media..
14:08:29.662          pjsua_media.c  ..RTP socket reachable at 10.0.0.229:4004
14:08:29.663          pjsua_media.c  ..RTCP socket reachable at 10.0.0.229:4005
14:08:29.663          pjsua_media.c  ..Media index 0 selected for audio call 1
14:08:29.663      udp0x7f40f4019410  ..UDP media transport created
...
14:08:29.793      inv0x7f40f4008578  ....SDP negotiation done: Success
14:08:29.793          pjsua_media.c  .....Call 1: updating media..
14:08:29.793          pjsua_media.c  .......Media stream call01:0 is destroyed
14:08:29.793      udp0x7f40f4019410  ......UDP media transport started
14:08:29.793            pjsua_aud.c  ......Audio channel update..
14:08:29.793     strm0x7f40f402fb68  .......VAD temporarily disabled
14:08:29.793      udp0x7f40f4019410  .......UDP media transport attached
14:08:29.793     strm0x7f40f402fb68  .......Encoder stream started
14:08:29.793     strm0x7f40f402fb68  .......Decoder stream started
14:08:29.793          pjsua_media.c  ......pjsua_aud_channel_update() failed for call_id 1 media 0: Invalid operation (PJ_EINVALIDOP)
14:08:29.793      udp0x7f40f4019410  .......UDP media transport detached
14:08:29.793          pjsua_media.c  .......Media stream call01:0 is destroyed
14:08:29.793      udp0x7f40f4019410  ......UDP media transport destroyed
14:08:29.793          pjsua_media.c  ......Error updating media call01:0: Invalid operation (PJ_EINVALIDOP)
14:08:29.793           pjsua_call.c  .....Unable to create media session: No active media stream after negotiation (PJMEDIA_SDPNEG_ENOMEDIA) [status=220048]
```

**Comment by cristeab** [comment]
Hi, when using PJSUA C API, one needs to use "pjsua_set_null_snd_dev();", after "pjsua_start();".

**Comment by razvancrainea** [comment]
Thanks for your response! I was actually calling that, but before `ep.libStart()`, as that's how I noticed it is done in pjsua:
```
  // Initialize endpoint                                                 
  MyEndpoint ep;                                                         
                                                                         
  ep.libCreate();                                                        
  EpConfig ep_cfg;                                                       
  ep.libInit(ep_cfg);                                                    
  ep.audDevManager().setNullDev(); 
   ...
  ep.libStart();
```
I tried to move it after `ep.libStart()`, but did not make any difference.

Note that if I start two calls in parallel (different processes), they work just fine. My problem, however, is that I need to start a second call from the same app, to be able to preserve the IP&port of the Endpoint.

**Comment by razvancrainea** [comment]
Not sure if it helps, but I think it is worth mentioning that the same thing happens when I get a REFER and transfer the current call to a new destination - when the second call starts and gets established with a 200 OK, I get the same error for the second call.

**Comment by wosrediinanatour** [comment]
I am not sure what you want to achieve. You are in a container, but want to have PJMEDIA provide a SDP?
SDP & media is done automatically by PJSUA2. If you don't need it you have to mock pjsua_aud.c (is done in an example), use your own SDPs via callback, and to skip allocating transport for RTP... (of course you can write your own pjsip-lib wrapper in C++).

**Comment by cristeab** [comment]
Please provide a working example showing this issue

**Comment by razvancrainea** [comment]
@wosrediinanatour no, I did not want to provide or modify the SDP whatsoever - all that PJSUA2 provides is enough for me. But my problem was that I was not able to have a second call in parallel with the first one.
After digging a bit more into this, I found that not the number of calls was the problem, but the fact that the calls were going to a media server. The answer/200 OK of this media server was actually rejected due to the fact that the pjmedia stack could not start resampling - this was my fault, because I compiled pjsua with resampling disabled. Why resampling is needed, I am not sure, as the established codec is PCMU with a 8000 rate, which is the default rate. However, re-enabling the resampling sorted out this issue, and I can now have the two call running in parallel.
@cristeab , @wosrediinanatour thank you very much for your time and help!
